src/docs/arch-hs.dox

```diff
@@ -1,8 +1,272 @@
+is the start timestamp for the update \c U1 and the stop timestamp in the value is the start
+timestamp for the update \c U2.
+
+When a modified history store table page is processed by reconciliation, certain records on the page
+are removed when creating a new page image if the tombstones on those records have become globally
+visible. This ensures that WiredTiger is only keeping the relevant historical data required to
+serve the oldest reader in the system or as dictated by the oldest timestamp set by the application.
+Once all records on a page are obsolete, the page itself can be removed to reduce the size of
+the history store table (see @ref arch-transaction for more details about oldest timestamp).
+
+@section arch_hs_read Searching for older versions of a key in History Store
+When looking for an update visible to the current transaction, WiredTiger first searches the update
+chain for any visible updates. If there is no visible update in the chain, WiredTiger then
+inspects the on-disk version of the key. If that version is not visible to the transaction,
+WiredTiger searches the history store table for a visible update for the key. When
+searching for update that satisfies the read timestamp constraint, WiredTiger starts with the newest
+update of the key in the history store table and then iterates through the older updates until there
+are no updates left to process. Note that although there can be multiple updates in the
+history store for a key and read timestamp combination, each update would have different
+visibility based on transaction and timestamp based visibility rules. In case there are no records
+visible in the history store table, \c WT_NOTFOUND error is returned to the reader.
+
+@section arch_hs_rts History tore and rollback-to-stable
+Rollback-to-stable searches the history store table for valid updates that are stable according to
+the supplied stable timestamp and replaces the on-disk version with an update from the history store
+table that fulfills the criteria. Rollback-to-stable also removes updates from the
+history store table that are not stable according to the stable timestamp and transaction snapshot
+being used for rollback-to-stable operation. More details about the rollback-to-stable operation
+can be found in the @ref arch-rts page.
+
+@section arch_hs_prepared History store and prepared transactions
+When there is a prepared update for a key and the page is evicted, the prepared update is written
+to the on-disk page and any older updates are written to the history store table. There can be
+no update from a different transaction that is newer than the prepared update for the key and
+therefore the history store should never contain a prepared update. When a prepared update is
+committed, the stop timestamp of the newest update in the history store table is updated from
+\c WT_TS_MAX to the commit timestamp of the prepared update. When prepared update is rolled back,
```

Comment:
You might be missing something here too:
`When *a* prepared updated`

Reply:
Fixed.